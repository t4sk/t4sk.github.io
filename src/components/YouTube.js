import React from "react"
import PropTypes from "prop-types"
import styles from "./YouTube.module.css"

export function getUrl({ url, start, end }) {
  let src = url
  const queryParams = {}

  if (start) {
    queryParams.start = start
  }
  if (end) {
    queryParams.end = end
  }

  const query = Object.entries(queryParams)
    .map(([key, val]) => `${key}=${val}`)
    .join("&")

  if (query) {
    return `${src}?${query}`
  }

  return src
}

export function YouTube(props) {
  // TODO responsive YouTube

  return (
    <div className={styles.component}>
      <iframe
        width="889"
        height="500"
        src={getUrl(props)}
        frameBorder="0"
        allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
        allowFullScreen
      ></iframe>
    </div>
  )
}

YouTube.propTypes = {
  url: PropTypes.string.isRequired,
  start: PropTypes.number,
  end: PropTypes.number,
}

export default YouTube
