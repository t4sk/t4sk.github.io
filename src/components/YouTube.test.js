import React from "react"
import { shallow } from "enzyme"
import { YouTube, getUrl } from "./YouTube"

const props = {
  url: "https://www.youtube.com/embed/Dh7r6Ze-0Bs",
  start: 123,
  end: 456,
}

beforeEach(() => {})

test("it renders", () => {
  const component = shallow(<YouTube {...props} />)

  expect(component).toMatchSnapshot()
})

test("getUrl", () => {
  expect(getUrl(props)).toEqual(
    `https://www.youtube.com/embed/Dh7r6Ze-0Bs?start=123&end=456`
  )
})
