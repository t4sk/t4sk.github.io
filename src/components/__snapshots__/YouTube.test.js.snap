// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div
  className="component"
>
  <iframe
    allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture"
    allowFullScreen={true}
    frameBorder="0"
    height="500"
    src="https://www.youtube.com/embed/Dh7r6Ze-0Bs?start=123&end=456"
    width="889"
  />
</div>
`;
